src/index.js

```diff
@@ -0,0 +1,183 @@
+const weatherGarden = document.getElementById('weather-garden');
+const weatherEmojis = document.getElementById('weather-emojis');
+const cityDisplay = document.getElementById("city-display")
+const cityId = document.getElementById("city-input")
+const skyOptions = document.getElementById('change-sky');
+const skyEmojis = document.getElementById('chosen-sky-emojis');
+const resetBtn = document.getElementById('resetbtn');
+
+
+
+
+
+
+function display() {
+    cityDisplay.innerText = cityId.value;
+}
+
+// console.log('curret temp', number.innerText);
+
+const incrementNum = () => {
+    state.number.innerText = state.temp++;
+}
+// color changing function
+const colorTemp = () => {
+
+    if (state.temp < 49){
+        state.number.style.color = "teal";
+    }
+    else if (state.temp < 59){
+        state.number.style.color = "green";
+    }
+    else if (state.temp < 69){
+        state.number.style.color = "yellow";
+    }
+    else if (state.temp < 79){
+        state.number.style.color = "orange";
+    }
+    else {
+
+        state.number.style.color = "red";
+    }
+
+}
+
+skyOptions.addEventListener('change', () => {
+  const skyValue = document.getElementById('change-sky').value;
+  if (skyValue == 'sunny') {
+    skyEmojis.innerText = '☀️☀️ ☁️ ☀️☁️ ☀️ ☁️☀️☀️';
+  } else if (skyValue =='cloudy') {
+    skyEmojis.innerText = '☁️☁️ ☁️ ☁️☁️ ☁️ 🌤 ☁️ ☁️☁️';
+  } else if (skyValue == 'rainy') {
+    skyEmojis.innerText = '🌧🌈⛈🌧🌧💧⛈🌧🌦🌧💧🌧🌧';
+  } else if (skyValue == 'snowy') {
+    skyEmojis.innerText = '🌨❄️🌨🌨❄️❄️🌨❄️🌨❄️❄️🌨🌨';
+  } else {
+    skyEmojis.innerText = '';
+  }
+})
+
+
+const weatherGardenChanges = () => {
+
+  if (state.temp < 59){
+      weatherGarden.style.backgroundColor = "green";
+      weatherEmojis.innerHTML = "🌲🌲⛄️🌲⛄️🍂🌲🍁🌲🌲⛄️🍂🌲"
+  }
+  else if (state.temp < 69){
+      weatherGarden.style.backgroundColor = "yellow";
+      weatherEmojis.innerHTML = "🌾🌾_🍃_🪨__🛤_🌾🌾🌾_🍃"
+  }
+  else if (state.temp < 79){
+      weatherGarden.style.backgroundColor = "orange";
+      weatherEmojis.innerHTML = "🌸🌿🌼__🌷🌻🌿_☘️🌱_🌻🌷"
+  }
+  else {
+          weatherGarden.style.backgroundColor = "red";
+          weatherEmojis.innerHTML = "🌵__🐍_🦂_🌵🌵__🐍_🏜_🦂"
+  }
+
+}
+
+
+upButton.addEventListener("click", () => {
+    console.log(state.temp);
+    incrementNum();
+    colorTemp(state.temp);
+    weatherGardenChanges(state.temp);
+});
+
+
+const decrementNum = () => {
+    state.number.innerText = state.temp--;
+}
+
+downButton.addEventListener("click", () => {
+    console.log(state.temp);
+    decrementNum()
+    colorTemp(state.temp)
+    weatherGardenChanges(state.temp);
+});
+
+const getLanLon = (city) => {
+    return axios.get(`http://localhost:5000/location?q=${city}`)
+    .then(response => {
+        let lat = response.data[0].lat
+        let lon = response.data[0].lon
+        console.log(response.data[0].display_name)
+        return [lat, lon]
+    })
+    .catch(error => {
+        console.log(error)
+    })
+}
```

Comment:
Here's an alternative to promise chaining! You can use the `async/await` keywords to have your function wait until a promise is fulfilled and then you pass the results of that promise to other pieces of your function like assigning your return values as variables or other things like that. It looks something like this:

```javascript
const findCityLocation = async () => {
    let lat;
    let lon;
    const resp = await axios.get('http://127.0.0.1:5000/location', {
        params: {
            q: document.getElementById("city").value
        }
    })
    
    try {
      [lat, lon] = resp.data[0]
      getWeather({ lat: lat, lon: lon});
    } catch (error) {
       console.log(error)
    }
};
```